### DISCUSSION TITLE
Riflenade bug?

### DISCUSSION THREAD

**Discussion by NeluZJ** [etlegacy/etlegacy#1930] [Q&A] [answered]
Hi guys,

Kimi sent me overhere to discuss/ask help about a bug/problem i got with a loaded rifle.

It is as following: 
I load a riflenade. After that i switch to luger or nade to spam. When switching back to rifle, the loaded rifle is sometimes gone. Sometimes it is still there. I even had that it was already loaded eventhough i never loaded the rifle.

I've recorded it aswell. 
You can see that in the first two tries it is gone as i subscribed above, the second time it dissapeard after the second nade, as you can check right side bottom (showing 1/4).
The third and fourth time i shot a few times with luger and it was still okay. (From 0.30 sec)
The fifth time you can cleary see that i get the riflenade out of the rifle (at 0.49 sec). However, after throwing one nade, you can see that i suddenly had a loaded rifle. I shot it, threw one more nade (did not load a new riflenade) and i still had one, as you can clearly see. One more nade, and the same! 

You can probably understand that this bug/problem is really annoying when spamming nades and want to shoot a riflenade. 

I've tried to change the cg_weapAltSwitches to "1" (my default) and "0" but both settings appear to have the same problem.
This is recorded on server  173 | nl.moestavern.pub;password goodluck
Other servers were the same.

I am wondering if some of my config settings are causing this problem. I dont know a lot about it, but maybe someone can check it?
[autoexec.txt](https://github.com/etlegacy/etlegacy/files/8254289/autoexec.txt)
https://www.youtube.com/watch?v=QHLwLYIzTfs for the video. It was to big to upload here :(

If anybody has a solution, i am glad to hear it!

Regard,
Krepox

**Comment by ryzyk-krzysiek** [comment]
I've investigated this bugs few months ago but I wasn't able to find a definitive solution to it. Figuring this out was enough confusing to me at the time to not remember exactly what was going on so explaning it would be even harder now. From what I remember there is few things at play:
- The way weapalt is handled
- The way reloading is handled
- weaponbank weapalt (also known as cg_weapAltSwitches)

Some (I believe) key code I vaguely remember:

Ammoclip issue:
https://github.com/etlegacy/etlegacy/blob/f731a7c18bd5e74251315114e62961f1750e5796/src/game/bg_pmove.c#L2789-L2801

Where the animation is cut out from playing:
https://github.com/etlegacy/etlegacy/blob/f731a7c18bd5e74251315114e62961f1750e5796/src/game/bg_pmove.c#L2640-L2644

There is also cgame code at play of course- `CG_AltWeapon_f` and `CG_WeaponBank_f` but I don't remember anything important from there.

This also isn't legacy specific (beside `cg_weapAltSwitches`), it is present in etpro (even worse to some degree as it can bug the weapon entirely quite easily it seems):
- https://streamable.com/6haqub etpro riflenade reload bug
- https://streamable.com/xmzxia  legacy riflenade reload bug (with cg_weapAltSwitches in the middle of video - loading a nade and it doesn't load)
- https://streamable.com/g5vubj riflenade reload bug with pistol (also in etpro)

To reproduce `cg_weapAltSwitches` bug simply press `weaponbank` bind during loading BUT if you successfully load it, it will stop happening till you shoot, because `ammoclip` is not 0 anymore, if I remember correctly.

To reproduce pistol/nade bug - after shooting/throwing press change to rifle and press weapalt or weaponbank again (if cg_weapAltSwitches 1).

Other weapalt stuff like silencer have similiar issues.

But not all seems to be lost, to fix those issues you can try the following:

- set `cg_weapAltSwitches 0`
- set `cg_weaponCycleDelay 0` afaik it doesn't fix it entirely but should (greatly?) help reduce the times the issue happens.
- try not to spam weapalt command too much if you can
- you can use these bugs to your advantage in the meantime :p

**Comment by ryzyk-krzysiek** [comment, marked as answer]
Just to note those issues were fixed here:
https://github.com/etlegacy/etlegacy/commit/b424741ef8ece630da18c154665227f7c7f862b7
